suprb/suprb.py

```diff
@@ -147,15 +147,27 @@ def fit(self, X: np.ndarray, y: np.ndarray, cleanup=False):
 
         # Fill population before first step
         if self.n_initial_rules > 0:
-            self._discover_rules(X, y, self.n_initial_rules)
+            try:
+                self._discover_rules(X, y, self.n_initial_rules)
+            except:
+                warnings.warn("Error in rule discovery first iteration")
+                return self
 
         # Main loop
         for self.step_ in range(self.n_iter):
             # Insert new rules into population
-            self._discover_rules(X, y, self.n_rules)
+            try:
+                self._discover_rules(X, y, self.n_rules)
+            except:
+                warnings.warn("Error in rule discovery first iteration")
```

Comment:
Change to:
"An error has occured when trying to discover rules." 

Additionally, please include the original error message

Reply:
Can we include the iteration without changing the for loop?

Reply:
I think we can, but I don't want to mess this up now. Maybe look into this if we want to refactor the code as a whole to have a clean code throughout the project?